pkg/operator/starter.go

```diff
@@ -126,9 +133,17 @@ func RunOperator(ctx context.Context, kubeconfig *restclient.Config) error {
 		configInformers.Config().V1().Images(),
 	)
 
+	var recorder events.Recorder
+	controllerRef, err := events.GetControllerReferenceForCurrentPod(context.TODO(), kubeClient, defaults.ImageRegistryOperatorNamespace, nil)
+	if err != nil {
+		klog.Warningf("Cannot create controller reference event recorder. Using logging event recorder: %v", err)
```

Comment:
This seems exit worthy.  Check to see what cluster-kube-apiserver-operator does.

Reply:
https://github.com/openshift/library-go/blob/4362aa519714a4b62b00ab8318197ba2bba51cb7/pkg/controller/controllercmd/builder.go#L230 seemed to allow for this without exiting, modeled it off of that.